be/src/Project/Services/Catalog/Catalog.Api/v1/Categories/Controllers/CategoriesController.cs

```diff
@@ -340,49 +354,53 @@ public async Task<IActionResult> SaveCategorySchema(CategorySchemaRequestModel r
         }
 
         /// <summary>
-        /// Gets schema by category id.
+        /// Gets schemas by category id.
         /// </summary>
         /// <param name="categoryId">The category id.</param>
         /// <returns>The category schema.</returns>
         [HttpGet, MapToApiVersion("1.0")]
         [Route("CategorySchemas/{categoryId}")]
         [AllowAnonymous]
-        [ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(CategorySchemaResponseModel))]
+        [ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(CategorySchemasResponseModel))]
         [ProducesResponseType((int)HttpStatusCode.NotFound)]
         [ProducesResponseType((int)HttpStatusCode.UnprocessableEntity)]
-        public async Task<IActionResult> GetCategorySchemaByCategoryId(Guid? categoryId)
+        public IActionResult GetCategorySchemasByCategoryId(Guid? categoryId)
         {
-            var serviceModel = new GetCategorySchemaServiceModel
+            var serviceModel = new GetCategorySchemasServiceModel
             {
-                CategoryId = categoryId,
+                Id = categoryId,
                 Language = CultureInfo.CurrentCulture.Name
             };
 
-            var validator = new GetCategorySchemaModelValidator();
+            var validator = new GetCategorySchemasModelValidator();
 
-            var validationResult = await validator.ValidateAsync(serviceModel);
+            var validationResult = validator.Validate(serviceModel);
 
             if (validationResult.IsValid)
             {
-                var categorySchema = await _categoryService.GetCategorySchemaAsync(serviceModel);
+                var categorySchemas = _categoryService.GetCategorySchemas(serviceModel).Result;
 
-                if (categorySchema != null)
+                if (categorySchemas != null)
```

Comment:
**suggestion (non-blocking):** I think you can change != to is not null anywhere